code/cmdline/cmdline.cpp

```diff
@@ -886,6 +892,11 @@ bool has_cmdline_only_flag(int argc, char *argv[])
 void os_init_cmdline(int argc, char *argv[])
 {
 	FILE *fp;
+
+#if defined(APPLE_APP)
+	char *path_name = SDL_GetBasePath();
+	SetCurrentDirectory(path_name);
```

Comment:
I think it would be better to initialize this in actual_main in `freespace.cpp` (somewhere around line 8750) which would match the location where the windows version does the same.